packages/nextjs/src/config/wrappers/withSentryGetServerSideProps.ts

```diff
@@ -8,8 +9,14 @@ import { wrapperCore } from './wrapperUtils';
  * @param route: The page's parameterized route
  * @returns A wrapped version of the function
  */
-export function withSentryGetServerSideProps(origGetServerSideProps: GSSP['fn'], route: string): GSSP['wrappedFn'] {
-  return async function (context: GSSP['context']): Promise<GSSP['result']> {
-    return wrapperCore<GSSP>({ origFunction: origGetServerSideProps, context, route, op: 'getServerSideProps' });
+export function withSentryGetServerSideProps(
+  origGetServerSideProps: GetServerSideProps,
+  route: string,
+): GetServerSideProps {
+  return function (...getServerSidePropsArguments: Parameters<GetServerSideProps>): ReturnType<GetServerSideProps> {
```

Comment:
Next has actual types for these, which is what's embedded in the hacky types objects I was using hoping I could get TS to be smart about the fact that they all went together. Any reason not to import them here, directly from next, and use them?

Reply:
It's just safer. Assume Next.js released that makes `getServerSideProps` accept an additional argument. The current implementation of the wrapper... breaks. It doesn't pass the new argument. The `Parameters` is the most elegant way to type this. I wouldn't change this.

Reply:
Okay, your call. I do see your point when it comes to the arguments, since you're gathering them into an array.